src/web/mjs/engine/Storage.mjs

```diff
@@ -869,4 +877,37 @@ export default class Storage {
         }
         return undefined;
     }
+
+    /**
+     * Save the given value for the given key in the bookmark storage
+     */
+    saveBookmarks( key, value, indentation ) {
+        return new Promise( ( resolve, reject ) => {
+            this.fs.writeFile( this.path.join(Engine.Settings.bookmarkDirectory.value, 'hakuneko.' + key), JSON.stringify( value, undefined, indentation ), function( error ) {
+                if( error ) {
+                    reject( error );
+                } else {
+                    resolve();
+                }
+            } );
+        } );
+    }
+
+    /**
+     * Load the value for the given key from the bookmark storage
+     */
+    async loadBookmarks( key ) {
+        return new Promise( ( resolve, reject ) => {
+            this.fs.readFile( this.path.join(Engine.Settings.bookmarkDirectory.value, 'hakuneko.' + key), 'utf8', ( error, data ) => {
```

Comment:
For consistency: Either use `this._bookmarkOutpotPath()` instead of `Engine.Settings.bookmarkDirectory.value` or keep as is and remove `this._bookmarkOutpotPath()`.